### ISSUE TITLE
Showing/Hiding Based on Visibility of Lines in Chart

### ISSUE THREAD

**Issue by wjohnsto** [chartjs/chartjs-plugin-datalabels#51]
I have a line chart with a legend and 3 lines drawn. When you click the legend the lines show/hide and the graph adjusts accordingly. Out of the box the datalabels do not hide whenever a line is hidden. I am currently using this method of hiding the datalabels:

```
options: {
    plugins: {
        datalabels: {
            display: (context: any) => {
                return !context.dataset._meta[0].hidden;
            },
        },
    },
},
```

This works, but the `_meta[0].hidden` thing seems like a hack/not something we're supposed to be using. Is there a different approach that is more stable?

**Comment by simonbrunel** [comment]
I think that's a bug, it should work out of the box.

Can you build a jsfiddle that reproduces this issue?

**Comment by hrobayo** [comment]
The same issue form me, the wjohnsto solutions works, Chart.js  Version: 2.7.2 used.

![datalabelerror](https://user-images.githubusercontent.com/6777410/39059277-b168a07a-4483-11e8-978d-0ae0d9da7616.png)

Please, is there another solution?

**Comment by simonbrunel** [comment]
That's a bug, will be fixed in next release (labels will always been hidden if the dataset is hidden).

As a workaround, you can use:

```javascript
display: (context: any) => {
    return context.chart.isDatasetVisible(context.datasetIndex);
}
```

**Comment by hrobayo** [comment]
It's nice!, thanks.

**Comment by wjohnsto** [comment]
Sweet 👍

**Comment by hoang051195** [comment]
Thanks

**Comment by hoang051195** [comment]
It's nice but how to add condition (ex: > 10).

**Comment by hoang051195** [comment]
Thanks. And I use this for display data with condition
`display: function(context) {
	if(context.chart.isDatasetVisible(context.datasetIndex)) {
		return context.dataset.data[context.dataIndex] > 0;  
	}
	return context.chart.isDatasetVisible(context.datasetIndex); 					
}`

**Comment by devinjhill** [comment]
Does anyone have the full solution??? I tried adding simonbrunel's solution and it breaks my page. I also tried adding his solution to wjohnsto's jsfiddle code and it broke that as well...

![devin_ipconfig_low](https://user-images.githubusercontent.com/24593229/42101700-d4b593fe-7b91-11e8-9082-fe104d7f728c.PNG)

**Comment by devinjhill** [comment]
Does anyone have the full solution??? I tried adding simonbrunel's solution and it breaks my page. I also tried adding his solution to wjohnsto's jsfiddle code and it broke that as well...

![chartjs_jsfiddle](https://user-images.githubusercontent.com/24593229/42101720-ea43f63e-7b91-11e8-8d39-374071c30c64.PNG)

**Comment by wjohnsto** [comment]
My code is TypeScript, so you may have to remove the non-JS stuff to get it to work.

**Comment by devinjhill** [comment]
@simonbrunel can you provide a pure js/chart.js work around to this?

**Comment by wjohnsto** [comment]
```
display: function (context) {
    return context.chart.isDatasetVisible(context.datasetIndex);
}
```

**Comment by simonbrunel** [comment]
@wjohnsto Thanks (I updated my comment)

**Comment by simonbrunel** [comment]
Fixed in 542ec921, will be released in 0.4.0
